### PR TITLE
add zero sum check to db-check

### PR DESCRIPTION
**Pull request by ryanpmortensen**

### CODE REVIEW

File: apps/workers/db-check/worker_test.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/hydronica/trial"
+	"github.com/jmoiron/sqlx"
+)
+
+func TestGetZeroSums(t *testing.T) {
+	type Row struct {
+		IntCol  int64
+		NumCol  []uint8
+		DateCol time.Time
```
Comment (jbsmith7741) [inline apps/workers/db-check/worker_test.go:17]: to improve readability you may want to use normal types and convert in the test function. 
``` go 
type Row struct {
date string // format of 2006-01-02T15
integer int
numeric string 
} 

```

File: apps/workers/db-check/worker_test.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+	type Expected struct {
+		ZR ZeroRecs
+		MR MissingRecs
+	}
+
+	d := time.Date(2023, 06, 15, 0, 0, 0, 0, time.UTC)
+	dt := make([]time.Time, 24)
+	for i := range dt {
+		dt[i] = d.Add(time.Hour * time.Duration(i))
+	}
```
Comment (jbsmith7741) [inline apps/workers/db-check/worker_test.go:35]: I'd moved this into the test case.

File: apps/workers/db-check/worker_test.go [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+	fn := func(in Input) (Expected, error) {
+		// setup mock db response
+		db, mock, _ := sqlmock.New()
+		eq := mock.ExpectQuery("select *")
+		column1 := mock.NewColumn("impression").OfType("INTEGER", int64(0))
+		column2 := mock.NewColumn("revenue").OfType("NUMERIC", []uint8("0.0"))
+		column3 := mock.NewColumn("date").OfType("TIMESTAMP", time.Time{})
+		rows := mock.NewRowsWithColumnDefinition(column1, column2, column3)
+		for i := range in.Rows {
+			rows.AddRow(in.Rows[i].IntCol, in.Rows[i].NumCol, in.Rows[i].DateCol)
```
Comment (jbsmith7741) [inline apps/workers/db-check/worker_test.go:46]:
``` go 
for _, r := range in.Rows {

rows.AddRow(int64(r.IntCal), []uint8(r.NumCol), trial.TimeHour(r.DateCol))
}

File: apps/workers/db-check/worker_test.go [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+		zr, mr, _ := w.GetZeroSums(context.Background())
+		return Expected{ZR: zr, MR: mr}, nil
+	}
+	cases := trial.Cases[Input, Expected]{
+		"timestamp_no_zero_missing": {
+			Input: Input{DateType: "ts", GroupTS: "", Rows: []Row{
+				{IntCol: int64(123), NumCol: []uint8("123.45"), DateCol: dt[0]},
+				{IntCol: int64(123), NumCol: []uint8("123.45"), DateCol: dt[1]},
+				{IntCol: int64(123), NumCol: []uint8("123.45"), DateCol: dt[2]},
```
Comment (jbsmith7741) [inline apps/workers/db-check/worker_test.go:67]: switch to easy to read format. 
`{123, "123.45", "2023-06-15T00"},`

File: apps/workers/db-check/worker_test.go (hunk 2 above)
Comment (ryanpmortensen) [reply apps/workers/db-check/worker_test.go:35]: Wouldn't it be better to do once, so that all test cases can use it?

**Review by jbsmith7741 — APPROVED** [review]
GetZeroSums would benefit from a little bit of refactoring, The unit tests look really good.

**Pull request merged**